### DISCUSSION TITLE
DNS not resolving

### DISCUSSION THREAD

**Discussion by TDUBMIKE** [redisson/redisson#5159] [Q&A] [answered]
We are attempting to use Redisson Tomcat Session Manager in an Azure App Service application with private endpoints. The list of DNS servers for the private endpoints are listed in the environment variable WEBSITE_DNS_SERVER_FROM_VNET. When Netty DefaultDnsServerAddressStreamProvider attempts to resolve the DNS it does not look in the environment variable for the list of DNS servers.  In Azure App Service the DNS servers for the private endpoints are not available from the OS.  I have this working fine in my development environment but when deployed to the Azure App Service the DNS cannot be resolved.  I get the following error.  Any thoughts on how to resolve this issue?

WARNING [main] io.netty.resolver.dns.DefaultDnsServerAddressStreamProvider.<clinit> Default DNS servers: [/8.8.8.8:53, /8.8.4.4:53] (Google Public DNS as a fallback)
SEVERE [main] org.apache.catalina.core.StandardContext.startInternal The session manager failed to start
	org.apache.catalina.LifecycleException: org.redisson.client.RedisConnectionException: Unable to connect to Redis server: QAT-Redis.privatelink.redis.cache.windows.net/20.225.132.15:6380

**Comment by mrniko** [comment, marked as answer]
You need to implement own `addressResolverGroupFactory` and override `DnsServerAddressStreamProvider` in `create()` method.

**Reply by TDUBMIKE** [reply]
Thank you for your reply.  Is this a known issue with Azure private endpoints?  Just wondering if someone has already done this and I could leverage their implementation.

**Reply by mrniko** [reply]
According to the APE documentation - https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-dns

`Use a private DNS zone. You can use [private DNS zones](https://learn.microsoft.com/en-us/azure/dns/private-dns-privatednszone) to override the DNS resolution for a private endpoint. A private DNS zone can be linked to your virtual network to resolve specific domains.`

Did you try it?

**Comment by TDUBMIKE** [comment]
Yes,  we are using the private DNS Zone as noted. This works and resolves fine when running from a local dev server. Our .NET apps are able to resolve the name just fine in the Azure App Service.  I'm not sure why the DNS is not being resolved for Redisson when using private endpoints in the Azure App Service.  Is there any way I can just provide (via configuration) the fall back servers to use, instead of falling back to Google? I didn't expect DNS to be the issue I faced.  If my only option is to implement my own addressResolverGroupFactory  I can look at doing that too.  I would rather not have any custom code, I would rather just use Redisson as is.
